app.py

```diff
@@ -108,15 +108,17 @@ def confirm_resource(resource_id, secret):
 
 @app.route('/libraries')
 @compressible
+@uses_location
```

Comment:
This is the same decorator the `/` route uses. It'll look in the `_location` query string for a latitude/longitude, and if that's not present it'll try IP geolocation.